src/controller/CHIPDeviceController.cpp

```diff
@@ -1614,6 +1614,20 @@ void OnBasicFailure(void * context, CHIP_ERROR error)
     commissioner->CommissioningStageComplete(error);
 }
 
+void NonConcurrentTimeout(void * context, CHIP_ERROR error)
+{
+    if (error == CHIP_ERROR_TIMEOUT)
+    {
+        ChipLogProgress(Controller, "Non-concurrent mode: Expected NetworkResponse Timeout, do nothing");
+    }
+    else
+    {
+        ChipLogProgress(Controller, "Non-concurrent mode: Received failure response %s\n", chip::ErrorStr(error));
+        DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);
+        commissioner->CommissioningStageComplete(error);
```

Comment:
This is a use-after-free.  We already did `CommissioningStageComplete`, and the DeviceCommissioner could well be deleted by now.  This function must not use `context` in any way, just log the unexpected situation.

Reply:
Updated.